fhir-search/src/main/java/com/ibm/fhir/search/parameters/QueryParameterValue.java

```diff
@@ -190,7 +208,7 @@ public String toString() {
                     throw new IllegalStateException("Components of a composite search parameter may only have a single value");
                 }
                 returnString.append(componentDelim).append(componentValues.get(0));
-                componentDelim = "$";
+                componentDelim = ofTypeModifier ? SearchConstants.PARAMETER_DELIMITER : SearchConstants.COMPOSITE_DELIMITER;
```

Comment:
where is QueryParameterValue.toString() used?
is it ok that this will produce `type.system|type.code|value` and not `type.system|type.code$value`?

Reply:
It is used when generating the self/next/previous links for the search response Bundle.

Reply:
awesome, then the use of `|` to match what came in makes perfect sense

Reply:
I'll update the comment to be more explicit about that.

Reply:
Updated.